src/bio/bio_device.c

```diff
@@ -438,19 +450,29 @@ fetch_pci_dev_info(struct nvme_ctrlr_t *w_ctrlr, const char *tr_addr)
 		return -DER_INVAL;
 	}
 
+	w_ctrlr->pci_cfg    = NULL;
+	w_ctrlr->pci_cfg_sz = 0;
+	if (fetch_pci_cfg) {
+		D_ALLOC(w_ctrlr->pci_cfg, NVME_PCI_CFG_SPC_MAX_LEN);
```

Comment:
Minor: looks to me this D_ALLOC should be done in alloc_ctrlr_info() along with other allocations, and this extra 'fetch_pci_cfg' parameter could be removed.

Reply:
My concern was I didn't want to be fetching this for standard Scans only when health is requested. If you think the overhead isn't worth worrying about then I can remove the fetch parameter and always populate it regardless of the scan type?

Reply:
removed adding fetch parameter

Reply:
Sorry, I didn't make it clear. I meant if you move the 'pci_cfg' allocation from this function to alloc_ctrlr_info(), then you don't need this extra 'fetch_pci_cfg' parameter for this function.

But I'm fine with totally removing this 'fetch_pci_cfg' too.

Reply:
I removed the flag so now the config space is populated on each dRPC scan but only processed in the control-plane if a health flag is specified in gRPC request.